es-ds-components/components/es-dropdown-select.vue

```diff
@@ -0,0 +1,242 @@
+<script setup lang="ts">
+import Dropdown from 'primevue/dropdown';
+
+interface Props {
+    disabled?: boolean;
+    label?: string;
+    modelValue?: any;
+    options?: string[] | { label: string; value: string }[];
+    placeholder?: string;
+    required?: boolean;
+    state?: boolean | null;
+}
+const props = withDefaults(defineProps<Props>(), {
+    disabled: false,
+    modelValue: undefined,
+    options: () => [],
+    placeholder: '',
+    label: '',
+    required: false,
+    state: null,
+});
+
+const slots = useSlots();
+const hasError = () => !!slots.errorMessage;
+
+const emit = defineEmits(['update:modelValue']);
+
+const isOpen = ref(false);
+const isFocused = ref(false);
+const isInputClicked = ref(false);
+const isLabelClicked = ref(false);
+
+const isSelected = (option: any) => option === props.modelValue;
+
+const id = useId();
+const labelId = useId();
+
+const blur = () => {
+    if (!isOpen.value && (isInputClicked.value || isLabelClicked.value)) {
+        isInputClicked.value = false;
+        isLabelClicked.value = false;
+    }
+    isFocused.value = false;
+};
+
+const hide = () => {
+    isOpen.value = false;
+
+    if (!isFocused.value) {
+        isInputClicked.value = false;
+        isLabelClicked.value = false;
+    }
+};
+</script>
+
+<template>
+    <div
+        class="input-wrapper justify-content-end"
+        :class="$attrs.class">
+        <label
+            :id="labelId"
+            :for="id"
+            :class="{ 'sr-only': label === '' }"
+            @click="isLabelClicked = true">
+            <span v-if="label !== ''">{{ label }}</span>
+            <span v-else>Select an option</span>
```

Comment:
Nit: these two lines could probably be simplified to `{{ label || 'Select an option' }}` without needing any `<span>` tags.